move/its/sources/its.move

```diff
@@ -417,6 +417,7 @@ fun test_register_coin() {
     let coin_management = its::coin_management::new_locked();
 
     register_coin(&mut its, coin_info, coin_management);
+    assert!(sui::event::events_by_type<its::events::CoinRegistered<COIN>>().length() == 1);
```

Comment:
```suggestion
    assert_event_len<its::events::CoinRegistered<COIN>>!(1);
```
create a macro fun `assert_event_len<$T>(len: u64)` in utils to make this readable and concise

Reply:
we should add event checks in gateway tests as well if we don't have it

Reply:
added fun (not macro). Event querying in tests is new in move, will add tests in gateway in new PR

Reply:
Yeah, separate PR for the gateway